lib/Conversion/TorchToTcp/Utils.cpp

```diff
@@ -133,6 +139,72 @@ Value torch_to_tcp::broadcast0DOr1DToNDAndMatchShape(
   return result;
 }
 
+SmallVector<int64_t> torch_to_tcp::getShapeFromPrimList(Operation *primListOp) {
+  auto listConstruct = dyn_cast<Torch::PrimListConstructOp>(primListOp);
+  assert(listConstruct && "Target must come from PrimListConstructOp");
+  SmallVector<int64_t> resultShape;
+  for (Value value : listConstruct.getElements()) {
+    int64_t num;
+    if (matchPattern(value, m_TorchConstantInt(&num)))
+      resultShape.push_back(num);
+    else
+      resultShape.push_back(ShapedType::kDynamic);
+  }
+  return resultShape;
+}
+
+Value torch_to_tcp::broadcast0DOr1DFromPrimList(
+    ConversionPatternRewriter &rewriter, Value input, Value target,
+    int64_t axisInOutput) {
```

Comment:
Instead of having this utility take in a `prim::ListConstruct` op, it would be more useful if it takes in a `ArrayRef<Value>`, which are already extracted from the list construct op. 

Same for the `getShapeFromPrimList` function above as well.

You could extract the input values into a vector in the caller function `matchAndRewrite`. This will also avoid having to check for `prim::ListConstruct` op in two places.

Reply:
I meant that in the callsite of the utility, we could do something like:
```
   auto resultShape = getShapeFromValues(primtListVals);
   auto resultOp = broadcast0DOr1D(rewriter, input, values, resultShape, axisInOutput);
```

This way, both the util funcs are not tied to a primList and can be used in other places as well.